dev/source/docs/building-px4-with-qtcreator.rst

```diff
@@ -0,0 +1,233 @@
+.. _building-px4-with-qtcreator:
+
+=====================================================================
+Building ArduPilot for Pixhawk/PX4 on Windows or Linux with QtCreator
+=====================================================================
+
+This article shows how you can set up Qt Creator for editing ArduPilot code
+and building for Pixhawk/PX4 targets on Windows and Linux.
+
+Preconditions for Linux
+=======================
+
+Follow the instructions in :ref:`Building for Pixhawk/PX4 on Linux with Make 
+<building-px4-for-linux-with-make>` to download the required source code 
+(*ardupilot*, *PX4Firmware* and *PX4NuttX*) and toolchain.
+
+Preconditions for Windows
+=========================
+
+Follow the instructions in :ref:`Pixhawk/PX4 on Windows with Make <building-px4-with-make>` 
+to download the required source code (*ardupilot*, *PX4Firmware* and *PX4NuttX*) and toolchain.
+
+Make sure you have no Cygwin installed (or have it at least out of the environment variables), 
+as this can get in the way of the PX4 toolchain.
+
+Install Qt Creator
+==================
+
+#. Download Qt and the Qt Creator IDE from the `Qt Creator website <https://www.qt.io/ide/>`__
+#. Follow the instructions to install the IDE on your platform.
+
+Run Qt Creator on Windows
+=========================
+
+We try to make sure that Qt Creator is running with all the necessary environment variables 
+that are needed to build an ArduPilot project.
+
+#. Go to your local PX4 toolchain directory and create a file called **qtcreator.sh**
```

Comment:
The .sh file is run by the shell script command tool (sh.exe), so I think it makes sense that it has the .sh extension. I don't know if it's necessary though.